File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) - 1
        print(right)
        maxW = 0
        while left <= right:
            print(f"[{left},{right}]")

            w = min(heights[right], heights[left]) * (right - left)
            maxW = max(maxW, w)

            if heights[right] < heights[left]: right -= 1
            else: left += 1

        return maxW

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.cpp
class Solution {
   public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() < 2) return 0;

        int left = 0, right = 1;

        int maxProfit = 0;

        while (right < prices.size()) {
            int profit = prices[right] - prices[left];

            if (profit < 0) {
                left = right;
            } else if (profit > maxProfit) {
                maxProfit = profit;
            }

            right++;
        }

        return maxProfit;
    }
};
